src/builder.rs

```diff
@@ -143,6 +143,16 @@ impl<'builder> Builder<'builder> {
         Ok(Keypair { private, public })
     }
 
+    /// Builds an asymmetric [`Keypair`] from a given secret key and specified dh algorithm.
+    pub fn build_keypair_from_secret_key(
+        &self,
+        secret_key: &'builder [u8],
```

Comment:
Should be able to use an anonymous lifetime here, no reason to tie it to `'builder`.